stdlib/private/SwiftPrivatePthreadExtras/PthreadBarriers.swift

```diff
@@ -73,13 +73,15 @@ public func _stdlib_pthread_barrier_init(
     return -1
   }
   barrier.pointee.mutex = UnsafeMutablePointer.allocate(capacity: 1)
-  if pthread_mutex_init(barrier.pointee.mutex!, nil) != 0 {
-    // FIXME: leaking memory.
+  guard pthread_mutex_init(barrier.pointee.mutex!, nil) == 0 else {
+    barrier.pointee.mutex!.deinitialize()
+    barrier.pointee.mutex!.deallocate(capacity: 1)
     return -1
   }
   barrier.pointee.cond = UnsafeMutablePointer.allocate(capacity: 1)
-  if pthread_cond_init(barrier.pointee.cond!, nil) != 0 {
-    // FIXME: leaking memory, leaking a mutex.
+  guard pthread_cond_init(barrier.pointee.cond!, nil) == 0 else {
+    barrier.pointee.cond!.deinitialize()
+    barrier.pointee.cond!.deallocate(capacity: 1)
```

Comment:
The mutex would also have to be cleaned up here. Maybe the whole thing belongs in a `defer` that just checks that `barrier.pointee.count` is non-zero?

Reply:
Or this can be merged into one guard statement:

```
guard pthread_mutex_init(barrier.pointee.mutex!, nil) == 0 && pthread_cond_init(barrier.pointee.cond!, nil) == 0 else {
  barrier.pointee.mutex!.deinitialize()
  barrier.pointee.mutex!.deallocate(capacity: 1)
  barrier.pointee.cond!.deinitialize()
  barrier.pointee.cond!.deallocate(capacity: 1)
  return -1
}
```

If one of them fails, both `mutex` and `cond` will be deallocated. What do you think?

Reply:
This still doesn't call {{pthread_mutex_destroy}} when {{pthread_mutex_init}} succeeds but {{pthread_cond_init}} fails.

Reply:
Understood. Updating code to destroy mutex when `pthread_cond_init` fails.